src-electron/db/query-upgrade.js

```diff
@@ -0,0 +1,89 @@
+/**
+ *
+ *    Copyright (c) 2020 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+const dbApi = require('./db-api.js')
+const queryPackage = require('./query-package.js')
+const queryAttribute = require('./query-attribute.js')
+const queryCluster = require('./query-cluster.js')
+const dbEnum = require('../../src-shared/db-enum.js')
+const fs = require('fs')
+const fsp = fs.promises
+
+/**
+ * Returns an array of objects containing global attributes that should be forced external.
+ *
+ * @export
+ * @param {*} db
+ * @param {*} attributeId
+ * @returns An array of objects
+ */
+
+async function getForcedExternalStorage(db, attributeId) {
+  let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
```

Comment:
is this just a package and not packages?
`let pkg = `

Reply:
what if we have a .zap file that requires Zigbee and Matter packages?